code/datums/actions/xeno_action.dm

```diff
@@ -169,14 +169,20 @@
 		button.color = "#f0b400c8" // rgb(240,180,0,200)
 	else
 		button.color = "#ffffffff" // rgb(255,255,255,255)
-
+	// remove and re-add to update it
+	button.cut_overlay(list(maptext_image))
+	button.add_overlay(list(maptext_image))
```

Comment:
xeno_actions dont use the PI for some reason this code breaks cooldown_image
```suggestion
	// remove and re-add to update it
	button.overlays -= maptext_image
	button..overlays += maptext_image
```
Additionally byond overlays use appearances for overlays so I am not sure this works, would be nice if you checked